managers/BaseManager.js

```diff
@@ -7,54 +7,134 @@ const config = require('../common/config');
 const eventmesh = require('../eventmesh');
 const CONST = require('../common/constants');
 const NotImplementedBySubclass = errors.NotImplementedBySubclass;
+const Conflict = errors.Conflict;
 
 class BaseManager {
 
-  registerWatcher() {
-    throw new NotImplementedBySubclass('registerWatcher');
+  registerWatcher(resourceGroup, resourceType, queryString) {
+    logger.debug(`Registering watcher for resourceGroup ${resourceGroup} of type ${resourceType} with queryString as ${queryString}`);
+    return eventmesh.apiServerClient.registerWatcher(resourceGroup, resourceType, this.handleResource, queryString)
+      .then(stream => {
+        logger.info(`Successfully set watcher with query string:`, queryString);
+        return setTimeout(() => {
```

Comment:
I would prefer if you use `Promise.delay` instead of `setTimeout` here so that we don't mix promises and global timers.